src/devices/device.ts

```diff
@@ -153,6 +208,38 @@ export class Device extends EventEmitter {
 
     public receive (message: Buffer) {
         this.notify("receive", { message });
+
+        const mode = this._mode;
+        if (mode === undefined) {
+            return;
+        }
+        const { name, raw, pct, si, values } = this._modes[mode];
+        const valueSize = Consts.ValueTypeSize[values.type];
```

Comment:
I haven't checked on all devices, but do all device modes only accept one type of value? Even for multiple values?

Are there any instances where a device could have ie. 3x int32, 1x uint8? I don't know if the UART protocol even supports that.

Reply:
I would say they mix everything as in the boost color and distance sensor:

```json
[
  {
    "name": "COLOR",
    "input": true,
    "output": false,
    "raw": { "min": 0, "max": 10 },
    "pct": { "min": 0, "max": 100 },
    "si": { "min": 0, "max": 10, "symbol": "IDX" },
    "values": {
      "count": 1,
      "type": 0
    }
  },
  {
    "name": "PROX",
    "input": true,
    "output": false,
    "raw": { "min": 0, "max": 10 },
    "pct": { "min": 0, "max": 100 },
    "si": { "min": 0, "max": 10, "symbol": "DIS" },
    "values": {
      "count": 1,
      "type": 0
    }
  },
  "[...]",
  {
    "name": "SPEC 1",
    "input": true,
    "output": false,
    "raw": { "min": 0, "max": 255 },
    "pct": { "min": 0, "max": 100 },
    "si": { "min": 0, "max": 255, "symbol": "N/A" },
    "values": {
      "count": 4,
      "type": 0
    }
  },
  "[...]"
]
```

For SPEC 1, which includes values from COLOR and PROX modes, it use 0-255 as range instead of 0-10 to match other values.

Reply:
The tilt sensor of the TechnicMediumHub has different data types for different modes in use (Int8, Int32, Int16) but within one mode, the data type is always the same. At least, that is what the device/spec is self-declaring.

````
Port: 99
    IOTypeId: TechnicMediumHubTiltSensor
    HardwareRevision: 0.0.0.1
    SoftwareRevision: 0.0.0.1
    OutputCapability: True
    InputCapability: True
    LogicalCombinableCapability: True
    LogicalSynchronizableCapability: True
    ModeCombinations: []
    UsedCombinationIndex: 0
    MultiUpdateEnabled: False
    ConfiguredModeDataSetIndex: []
    Mode: 0
      Name: POS
      IsInput: True
      IsOutput: False
      RawMin: -180
      RawMax: 180
      PctMin: -100
      PctMax: 100
      SIMin: -180
      SIMax: 180
      Symbol: DEG
      InputSupportsNull: False
      InputSupportFunctionalMapping20: True
      InputAbsolute: True
      InputRelative: False
      InputDiscrete: False
      OutputSupportsNull: False
      OutputSupportFunctionalMapping20: False
      OutputAbsolute: False
      OutputRelative: False
      OutputDiscrete: False
      NumberOfDatasets: 3
      DatasetType: Int16
      TotalFigures: 3
      Decimals: 0
      DeltaInterval: 0
      NotificationEnabled: False
    Mode: 1
      Name: IMP
      IsInput: True
      IsOutput: False
      RawMin: 0
      RawMax: 100
      PctMin: 0
      PctMax: 100
      SIMin: 0
      SIMax: 100
      Symbol: CNT
      InputSupportsNull: False
      InputSupportFunctionalMapping20: False
      InputAbsolute: False
      InputRelative: True
      InputDiscrete: False
      OutputSupportsNull: False
      OutputSupportFunctionalMapping20: False
      OutputAbsolute: False
      OutputRelative: False
      OutputDiscrete: False
      NumberOfDatasets: 1
      DatasetType: Int32
      TotalFigures: 3
      Decimals: 0
      DeltaInterval: 0
      NotificationEnabled: False
    Mode: 2
      Name: CFG
      IsInput: False
      IsOutput: True
      RawMin: 0
      RawMax: 255
      PctMin: 0
      PctMax: 100
      SIMin: 0
      SIMax: 255
      Symbol:
      InputSupportsNull: False
      InputSupportFunctionalMapping20: False
      InputAbsolute: False
      InputRelative: False
      InputDiscrete: False
      OutputSupportsNull: False
      OutputSupportFunctionalMapping20: False
      OutputAbsolute: True
      OutputRelative: False
      OutputDiscrete: False
      NumberOfDatasets: 2
      DatasetType: SByte
      TotalFigures: 3
      Decimals: 0
      DeltaInterval: 0
      NotificationEnabled: False
````